db/migrate/20170326212208_create_sponsored_posts.rb

```diff
@@ -4,6 +4,7 @@ def change
       t.string :title
       t.text :body
       t.integer :price
+      t.references :topic, index: true
```

Comment:
This is the critical missing line, it establishing the relationship between the two tables.